shanoir-ng-studies/src/main/java/org/shanoir/ng/study/service/StudyServiceImpl.java

```diff
@@ -325,10 +330,21 @@ public Study update(Study study) throws ShanoirException {
 
 			for (SubjectStudy subjectStudyDb : studyDb.getSubjectStudyList()) {
 				if(!updatedIds.contains(subjectStudyDb.getId())) {
-					removed.add(subjectStudyDb.getSubject());
+					Subject sub = subjectStudyDb.getSubject();
```

Comment:
ok so it works only when you delete a whole study

shouldn't this be in subject service as well ?